featurebyte/service/observation_table.py

```diff
@@ -184,6 +186,10 @@ async def validate_materialized_table_and_get_metadata(
                 f"Point in time column should have timestamp type; got {point_in_time_dtype}"
             )
 
+        # Check that there's at least one entity mapped in the columns info
+        if not any(info.entity_id is not None for info in columns_info):
+            raise ValueError("At least one entity column should be provided.")
```

Comment:
Observation table without any entity is valid for for features created without entity, though it should be rare that a feature list only contains such features. Curious what is the rationale of enforcing this new rule?

Reply:
I think the validation makes sense as it avoids potential user error in most cases other than the edge case I mentioned. Is this validation applicable for `BatchRequestTable` as well?

Reply:
as discussed in person, reason for implementing this is to ensure the observation tables have some entity column that can be used for serving. 

this is only for OT's created from source tables, views and targets.